rotkehlchen/premium/sync.py

```diff
@@ -269,25 +291,14 @@ def try_premium_at_start(
             return None
 
         result = self._can_sync_data_from_server(new_account=create_new)
-        if result.can_sync == CanSync.ASK_USER:
```

Comment:
no no no. You are changing functionality here. By removing this code and putting it only under create_new_user you are changing the already established functionality. Please do not do that.

The logic is dead simple.

### Before your PR

If a user logs in or is created **and have premium** then the sync logic happens. How this manifests is for new user creation automatically the saved user DB is pulled. For login, you get asked (this is the sync_approval part) if you would like to change your local with the remote.

## After your PR

For login the same exact thing should happen.

For user creation, the api should pass an extra argument which will determine if the download logic should happen or not.

Reply:
My bad. You move it also in the else code and that's not easily visible in github diff.